### PR TITLE
edgedb authenticate

### PR DESCRIPTION
**Pull request by fantix**

### CODE REVIEW

File: src/server/authenticate.rs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+                '0'..='9' => x,
+                _ => '_',
+            }).collect::<String>();
+            q.default(&default);
+            credentials::path(&q.ask()?)
+        }
+    }?;
+    if cred_path.exists() {
+        if cmd.assume_yes {
+            eprintln!("{} will be overwritten!", cred_path.display());
```
Comment (tailhook) [inline src/server/authenticate.rs:144]: I would say `Owerwriting {filename}`. Since "will be" might be interpreted as "would be".

**Review by tailhook — COMMENTED** [review]
Good start. But we also want:
1. Asking all credentials, that aren't specified in command-line (host, user, password)
2. Authorize the actual connection, to verify that credentials are good.

**Comment by fantix** [comment]
Thanks for the review! I pushed a CRF and hopefully addressed most concerns.

> Authorize the actual connection, to verify that credentials are good.

`builder.connect_with_cert_verifier()` should've done it.

File: src/server/authenticate.rs [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                // `self.prompt == true` means no cert was provided to the CLI,
+                // in which case we should check the hostname
+                if self.verify_hostname.unwrap_or(self.prompt) {
+                    cert.verify_is_valid_for_dns_name(dns_name)
+                        .map_err(TLSError::WebPKIError)?;
+                }
+            }
+            Err(e) => {
+                // Bail out if we shouldn't ask the user to trust any cert
+                if !self.prompt {
```
[the remark this answers is not in the document]
Comment (fantix) [reply src/server/authenticate.rs]: I'm not sure I quite get the idea here - it's probably my bad to name it as `prompt`. In the CRF I renamed the variable to `system_ca_only` and added more comments.

[the remark this answers is not in the document]
Comment (tailhook) [reply src/server/authenticate.rs]: I mean if `self.prompt` is false, why do use `InteractiveCertVerifier`, when you could have used `edgedb_client::tls::CertVerifier` and achieve the same result?

[the remark this answers is not in the document]
Comment (fantix) [reply src/server/authenticate.rs]: Ah, that makes sense! Let me see here

**Pull request closed without merging**